File: Visitor/python-source/rooms/Kitchen.py
from __future__ import annotations

from typing import TYPE_CHECKING

from .ElementRoom import ElementRoom

if TYPE_CHECKING:
    from workers import Visitor


class Kitchen(ElementRoom):
    def accept(self, visitor: Visitor) -> None:
        visitor.visit_kitchen(self)

    def get_name(self) -> str:
        return "Kitchen"

File: JavaSource/src/main/java/ru/akhcheck/patterns/visitor/rooms/Kitchen.java
package ru.akhcheck.patterns.visitor.rooms;

import ru.akhcheck.patterns.visitor.workers.Visitor;

/** Кухня: сюда приходят все рабочие, кроме оконщика в санузел. */
public class Kitchen extends ElementRoom {
    @Override
    public void accept(Visitor visitor) {
        visitor.visit(this);
    }

    @Override
    public String getName() {
        return "Kitchen";
    }
}
